zettarepl/transport/base_ssh.py

```diff
@@ -33,11 +33,15 @@ def run(self):
     def wait(self):
         self.logger.debug("Waiting for exit status")
         exitcode = self.stdout_fd.channel.recv_exit_status()
-        try:
-            stdout = self.stdout_fd.read().decode(self.encoding)
-        except IOError as e:
-            self.logger.debug("Unable to read stdout: %r", e)
-            stdout = ""
+        if self.stdout is None:
```

Comment:
`self.stdout` is a `queue.Queue` (or `None` by default) to which we stream command output in a separate thread. If it is specified, we should not read subprocess pipe or SSH channel in `wait()` because it'll be a race condition of two readers reading one stream.